sfklCoding.cpp

```diff
@@ -305,6 +290,37 @@ int ReadHeader(V2_FILEHEADER *FileHeader, BYTE *fbuf, int bufsize)
   return SFARKLIB_SUCCESS;
 }
 
+// =================================================================================
+void InitFilenames(const char *OrigFileName, const char *InFileName, const char *ReqOutFileName)
+{
+	if (ReqOutFileName)
+	{
+		ChangeFileExt(ReqOutFileName, SfExt, OutFileNameMain, sizeof(OutFileNameMain));
+		ChangeFileExt(ReqOutFileName, NotesExt, OutFileNameNotes, sizeof(OutFileNameNotes));
+		ChangeFileExt(ReqOutFileName, LicenseExt, OutFileNameLicense, sizeof(OutFileNameLicense));
+	}
+	else
+	{
+		if (strchr(OrigFileName, '/') || strchr(OrigFileName, '\\'))
+		{
+			msg("Please run Decode again and provide an output filename. We cannot use the original filename because"
+				" it contains a disallowed character.", MSG_PopUp);

```

Comment:
Yes, great!

Reply:
Glad this was approved, I wanted to cut off any potential directory traversal attacks.